openassessment/xblock/staff_area_mixin.py

```diff
@@ -370,8 +364,24 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
             submission = submissions[0]
             submission_uuid = submission['uuid']
 
+        # This will add submission (which may be None) and username to the context.
         context = self.get_student_submission_context(student_username, submission)
 
+        # Only add the rest of the details to the context if a submission exists.
+        if submission_uuid:
+            self.add_submission_context(submission_uuid, context)
+
+        path = 'openassessmentblock/staff_area/oa_student_info.html'
+        return path, context
+
+    def add_submission_context(self, submission_uuid, context):
+        """
+        Add the submission information (self asssessment, peer assessments, final grade, etc.)
+        to the supplied context for display in the "learner info" portion of staff tools.
+        Args:
+            submission_uuid (unicode): The uuid of the submission, should NOT be None.
+            context: the context to update with additional information
+        """
         assessment_steps = self.assessment_steps
 
         example_based_assessment = None
```

Comment:
The cause of the bug is in the collapsed code below.

workflow = self.get_workflow_info(submission_uuid=submission_uuid) will return the workflow associated with self.submission if submission_uuid is None. Therefore, if the staff member viewing staff tools had submitted an assessment, workflow would be for the staff member, not the student being searched for.

And if the staff member had a grade on that assessment, grade_exists would be True. With those conditions, code blocks that should not have been entered would be entered.